feat: add volume patches to mfe service and k8s

Ang-m4 opened the pull request:
## Description

This pull request enhances the `tutor-mfe` plugin by adding the necessary hooks to allow other plugins, such as `tutor-paragon`, to serve static files using the `mfe` Caddy service. This is achieved by introducing new volume patches for both local and Kubernetes deployments and updating the service logic for development environments.

> [!NOTE]
> For further context, please refer to the following discussion: [https://github.com/openedx/openedx-tutor-plugins/pull/38](https://github.com/openedx/openedx-tutor-plugins/pull/38)


### Key Changes

* **Volume Support for MFE Service**:
    * A new `mfe-volumes` patch has been added to the local Docker Compose service to allow mounting additional volumes.
    * A corresponding `mfe-k8s-volumes` patch has been added to the Kubernetes deployment configuration.

* **Development Environment Enhancements**:
    To support hosting external files even when no MFEs are running in production mode, the `dev` service logic has been updated:
    1.  A default port mapping (`8002:8002`) is now included to ensure the `mfe` Caddy container is always accessible from the host.
    2.  The conditional check for starting the `mfe` service now evaluates `or PARAGON_VERSION is defined`. This handles the edge case where a user mounts all available MFEs for development (`mfe_data.unmounted` is empty), but still needs the `mfe` service to host custom files like those from Paragon.
* **Documentation**:
     The `README.rst` has been updated to include documentation for the new `mfe-volumes` and `mfe-k8s-volumes` patch points.

brian-smith-tcril commented on tutormfe/patches/local-docker-compose-dev-services:
```diff
@@ -20,9 +20,10 @@
 
 
-{% if mfe_data.unmounted|length > 0 %}
+{% if mfe_data.unmounted|length > 0 or PARAGON_VERSION is defined %}
```
Could you explain where `PARAGON_VERSION` is being defined?

brian-smith-tcril replied:
I think that having this functionality (the ability to tell this service that it should still host things even if we don't have any MFE bundles to host) is good functionality to add, and doesn't need to be couple to Paragon itself. I think changing this to be something like `STILL_HOST_THINGS_PLEASE` and having the `tutor-contrib-paragon` plugin set that to true when we want to host styles would be ideal.

DawoudSheraz reviewed (commented):
Let's create this PR against release branch. main branch is supposed to contain changes for next release, breaking changes, etc. This is a new feature that does not impact any existing functionality. We can add it in release so anyone using the current tutor version (V20) can also benefit from it.

brian-smith-tcril commented on tutormfe/patches/local-docker-compose-dev-services:
```diff
@@ -20,9 +20,10 @@
 
 
-{% if mfe_data.unmounted|length > 0 %}
+{% if mfe_data.unmounted|length > 0 or MFE_HOST_EXTRA_FILES %}
```
If there are no mounted MFEs that means that all the MFEs are being served via their static bundles. Those static bundles can still reference other statically hosted files.

The use case that led to the creation of this PR comes from https://github.com/openedx/openedx-tutor-plugins/pull/38. With design tokens, it is possible to set `PARAGON_THEME_URLS` in `MFE_CONFIG` to enable changing themes at runtime without rebuilding the MFEs.

One recommended way to do so is by hosting the theme `.css` files on a CDN like `jsdelivr` ([as 2U is doing for the elm theme for edx.org](https://www.jsdelivr.com/package/npm/@edx/elm-theme?tab=files)).

By adding the ability to host additional static files to the `tutor-mfe` caddy configuration, the runtime theming functionality can also be accomplished without using an external CDN, as `PARAGON_THEME_URLS` can be configured to point to URLs coming from caddy within the `tutor-mfe` plugin.

DawoudSheraz replied:
Hmmm. The usage of the config here is to have that port expose statement, even if the unmounted length is 0, right?

Ang-m4 replied:
@DawoudSheraz .
The conditional logic was extended to cover the case where no MFEs are unmounted (`mfe_data.unmounted` is empty).

This ensures the `mfe` service still starts and exposes port 8002, allowing it to serve external static files, for example, from Paragon, even in development environments without active MFEs. 

When this is needed, we can externally update `MFE_HOST_EXTRA_FILES` to make them available.

DawoudSheraz replied:
@Ang-m4 Right. Let's update the readme where we explain MFE_HOST_EXTRA_FILES to add this additional and necessary context.

arbrandes reviewed (approved):
I'm approving this, but will hold off merging until @DawoudSheraz has no more concerns.

brian-smith-tcril commented:
@Ang-m4 - based on @DawoudSheraz's latest comment it sounds like the logic in this PR isn't quite right. It was my understanding that we *only* wanted to host the extra files when `MFE_HOST_EXTRA_FILES` was set, and it seems that is not what the code is doing.

Ang-m4 commented:
I’ve just adjusted the logic so that hosting extra files in all environments is always controlled by the `MFE_HOST_EXTRA_FILES` setting. This way, the files will only be served when the setting is explicitly enabled, and in development it will also expose port `8002` for direct access.

I also updated the README to reflect this behavior more clearly, so it now matches what the setting actually does.

DawoudSheraz commented:
> Hi @DawoudSheraz @ahmed-arb could you let us know what the next steps are in the review process? We're eager to finish out the project of supporting design tokens in Tutor.

Hi. This will be merged as soon as @ahmed-arb completes the review.
